pyk/src/pyk/cterm/symbolic.py

```diff
@@ -212,11 +212,11 @@ def implies(
         except SmtSolverError as err:
             raise self._smt_solver_error(err) from err
 
-        if not result.satisfiable:
+        if not result.valid:
             if result.substitution is not None:
-                _LOGGER.debug(f'Received a non-empty substitution for unsatisfiable implication: {result.substitution}')
+                _LOGGER.debug(f'Received a non-empty substitution for non-valid implication: {result.substitution}')
```

Comment:
```suggestion
                _LOGGER.debug(f'Received a non-empty substitution for falsifiable implication: {result.substitution}')
```

Reply:
By the way, whats the expected relationship between `valid` being `True` and `substitution` / `predicate` being `None`?

Reply:
I believe it's the following:
- if implication is valid, backend must return a non-none substitution and predicate
- otherwise, backend may return non-none substitution or predicate, but they will be ignored (with warnings) and the substitution coming back from `CTermSymbolic.implies` will be set to `None`

Reply:
I first thought we would possibly omit empty substitutions/predicates, but indeed the endpoint returns 
`substitution: #Top` and `predicate: #Top` in those cases.